File: core/repository/runtime_topology_engine.py
from __future__ import annotations

from typing import Dict, List

from core.parsers.runtime_resolution_engine import resolve_runtime


def infer_runtime_topology(dependencies: List[str], imports: List[str] | None = None) -> Dict[str, object]:
    runtime = resolve_runtime(dependencies, imports or [])
    return {
        "runtimes": runtime.get("runtimes", []),
        "frameworks": runtime.get("frameworks", []),
        "api_indicators": runtime.get("api_indicators", []),
        "evidence": "parser_runtime_resolution",
    }

File: src/repository/runtimeTopologyEngine.ts
/**
 * Converted from Python: core/repository/runtime_topology_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { resolveRuntime } from "../parsers/runtimeResolutionEngine.js";

export function inferRuntimeTopology(dependencies: any, imports: any = null): any {
  var runtime: any = resolveRuntime(dependencies, py.or2(imports, () => ([])));
  return {"runtimes": py.get(runtime, "runtimes", []), "frameworks": py.get(runtime, "frameworks", []), "api_indicators": py.get(runtime, "api_indicators", []), "evidence": "parser_runtime_resolution"};
}
export { resolveRuntime };
